test/trusted_resources_test.go

```diff
@@ -0,0 +1,375 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/pod"
+	"github.com/tektoncd/pipeline/test/parse"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/system"
+	knativetest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/helpers"
+)
+
+var (
+	neededFeatureFlags = map[string]string{
+		"resource-verification-mode": "enforce",
+		// Make sure this is running under alpha integration tests
+		"enable-api-fields": "alpha",
+	}
+	privKey  = "trustedresources-keys/cosign.key"
+	pubKey   = "trustedresources-keys/cosign.pub"
+	password = "1234"
+)
+
+func init() {
+	os.Setenv("PRIVATE_PASSWORD", password)
+}
+
+func TestTrustedResourcesVerifyTask(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	c, namespace := setup(ctx, t, requireAnyGate(neededFeatureFlags))
```

Comment:
apologies for the potential scope creep here, but i want to double check that setting these flags as part of this test won't impact any other tests, do you know if that's the case? i'm wondering specifically two things:
- do we run the e2e tests in parallel? if so, do we do anything to make sure that other tests aren't impacted by controller wide config settings like this?
- if the test gets interrupted, will teardown occur that will reset these feature flags? (i would expect that the function being called by `CleanupOnInterrupt` would do something to reset these?)

In any case it's probably outside of the scope of this PR to do anything about these but i want to make sure that if these need to be addressed we do something to track that - BUT if these are already addressed, we should at least document how this works somewhere (maybe in a docstring for the `setup` function?) - or if there are already docs for this feel free to just point me at them 🙏

Reply:
Yes that is the tricky part of the tests. 

1) This setup function doesn't set the feature flag, it will only check if the needed flags are enabled or not, if not it will skip the current test. So here since we require alpha feature flag, the test will only run in alpha tests.
```
c, namespace := setup(ctx, t, requireAnyGate(neededFeatureFlags))
```
2) In the current PR we turn on `resource-verification-mode` by making api request to the api server and will turn it off after the test finished. It seems working because the CI passes and we're not running them in parallel (no `-parallel`).  
https://github.com/tektoncd/pipeline/blob/55459dd85258e47da484208d92d9b616633ce230/test/e2e-tests.sh#L83
I can add comment here that this would be trouble if we run tests in parallel.

I cannot think of a good solution here since this might be the first case that one feature flag may break others. It may not be worth atm to add a new CI job.

Reply:
> This setup function doesn't set the feature flag, it will only check if the needed flags are enabled or not, if not it will skip the current test.

ah, kk, thanks for explaining!

> I can add comment here that this would be trouble if we run tests in parallel.

kk sounds good - it would be nice if we could actually fail the test if someone tries to do it (if this went wrong, tests might just start randomly failing and it might be hard to trace that back to this comment) but I'm not sure how to do that (probably possible tho?)

Reply:
It should be not so hard to trace, since the error message will tell that the error is because we enable this feature flag. I think maybe I should raise this to the community and seek more feedback?